ontology/uco-gufo.ttl

```diff
@@ -20,6 +21,15 @@
 		;
 	.
 
+drafting:Bearing
+	a
+		gufo:Kind ,
```

Comment:
A Bearing, in the meaning of Direction / Orientation, indeed is a Quality. 

It is *not*, however, a Kind - these reflect rigid sortals. As a Quality, it is existentially dependent on Kinds. 

Since a Bearing can only occur when something is moving, it must therefore inhere_in a Kind that represents a moving object. This inherence relation is absent here and should be added.

Reply:
Also, I did already have the inherence between the quality instances and the vehicle:

```turtle
kb:Bearing-74c2bb20-1e4f-40ac-807e-bc2c528e39da
	a drafting:Bearing ;
	gufo:inheresIn kb:Vehicle-c7e910e8-4bed-473e-ad74-3df3eb63cbcc ;
	.
kb:Velocity-62bc6232-b570-45d7-b382-d4e0a36fceb9
	a drafting:Velocity ;
	gufo:inheresIn kb:Vehicle-c7e910e8-4bed-473e-ad74-3df3eb63cbcc ;
	.
kb:Vehicle-c7e910e8-4bed-473e-ad74-3df3eb63cbcc
	a drafting:Vehicle ;
	.
```

Was there another inherence matter you meant?

Reply:
> The property that led me to consider `gufo:Kind` is `gufo:concernsQualityType`, which has a range of `gufo:EndurantType`. I've looked a couple times thinking that it wanted an instance of the endurant type, rather than the endurant type itself, but, no, it's asking for the endurant type.

I'm not sure. Look at the [documented example](https://nemo-ufes.github.io/gufo/#situations):
```
:JohnWeighs80Kgin2015 rdf:type gufo:QualityValueAttributionSituation ;
                    gufo:concernsQualityType :Mass ;
                    gufo:concernsQualityValue "80.0"^^xsd:double ;
                    gufo:hasBeginPointInXSDDate "2015-01-01"^^xsd:date ;
                    gufo:hasEndPointInXSDDate "2015-12-31"^^xsd:date .
```

and consider the [earlier definition](https://nemo-ufes.github.io/gufo/#basicqualities) on `:Mass`:
```
:Mass rdf:type owl:Class ;
            rdfs:subClassOf gufo:Quality .
```

Then we can only conclude that `:Mass`, a class that is *only* subclassed from `gufo:Quality`, is considered a correct range for `gufo:concernsQualityType`. No need to state that `:Mass` is of a particular `gufo:EndurantType`.

Reply:
> [...] No need to state that `:Mass` is of a particular `gufo:EndurantType`.

This need arises from the `rdfs:range` of `gufo: concernsQualityType`: `gufo:Quality` is not, itself, typed as a `gufo:EndurantType`.  I suspect this stems from the disjoint-unions encoding `gufo:EndurantType`.

Reply:
> I've looked a couple times thinking that [`gufo:concernsQualityType`] wanted an instance of the endurant type, rather than the endurant type itself, but, no, it's asking for the endurant type.

The `rdfs:range` property doesn't *want* or *ask* anything; that's the shacl data mode of thinking. Remember that the `rdfs:range` property is not enforcing the range of a property, but an indication for inducing that the class in range of such property is of the indicated type. You're on the "describing reality" plane here, not the "prescribing data structure" plane.

Reply:
> > I've looked a couple times thinking that [`gufo:concernsQualityType`] wanted an instance of the endurant type, rather than the endurant type itself, but, no, it's asking for the endurant type.
> 
> The `rdfs:range` property doesn't _want_ or _ask_ anything; that's the shacl data mode of thinking. Remember that the `rdfs:range` property is not enforcing the range of a property, but an indication for inducing that the class in range of such property is of the indicated type. You're on the "describing reality" plane here, not the "prescribing data structure" plane.

I disagree, and I think there have been other efforts that disagree.  `rdfs:range` _does_ cause a demand on the type of the linked node, via [RDFS entailment](https://www.w3.org/TR/rdf11-mt/#patterns-of-rdfs-entailment-informative), specifically pattern `rdfs3`.

Other efforts that wish to express property ranges (and likewise with domains), but not necessarily to induce entailment behaviors, include Dublin Core's [`dcam:rangeIncludes`](https://www.dublincore.org/specifications/dublin-core/dcmi-terms/#http://purl.org/dc/dcam/rangeIncludes) and Schema.org's [`schema:rangeIncludes`](https://schema.org/rangeIncludes), to my knowledge.

The SHACL shapes across the CDO shape and profile repositories follow the pattern that if `rdfs:range` is used, then the linked thing should be conformant with the range, after entailment is completed.  A light form of entailment (a halting-guaranteed subset of the RDFS rules; OWL rules coming later) is in [`src/entail.py`](https://github.com/Cyber-Domain-Ontology/CDO-Shapes-Example/blob/2a1c4166f09c578e3345a5f922ac8dd33535be06/src/entail.py) , and that script is used in each of the shape and profile repositories.

So, between the "describing reality" plane, and the "prescribing data structure" plane, the shape and profile repositories check the reality prescriptions from `rdfs:range`.

Reply:
We've concluded that we've got two options:

1. Accept that gUFO requires that a Quality is also a ConcreteIndividual, and proceed accordingly, i.e., leave line 26 as it is now;
2. Anticipate the we have run into an error with gUFO, and conclude that when we remove line 26 we remain compliant to (a corrected and future release of) gUFO.

Since option 1 is safer than option 2, we proceed as in option 2.